homeassistant/components/energyzero/diagnostics.py

```diff
@@ -0,0 +1,58 @@
+"""Diagnostics support for EnergyZero."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import EnergyZeroDataUpdateCoordinator
+from .const import DOMAIN
+from .coordinator import EnergyZeroData
+
+
+def get_gas_price(data: EnergyZeroData, hours: int) -> float | None:
+    """Get the gas price for a given hour.
+
+    Args:
+        data: The data object.
+        hours: The number of hours to add to the current time.
+
+    Returns:
+        The gas market price value.
+    """
+    if not data.gas_today:
+        return None
```

Comment:
Want to add a test case to cover this line?

Reply:
Yes I would, but I just couldn't figure it out right now

Reply:
```diff
diff --git a/tests/components/energyzero/test_diagnostics.py b/tests/components/energyzero/test_diagnostics.py
index 0871975aae..e58ca9bc0b 100644
--- a/tests/components/energyzero/test_diagnostics.py
+++ b/tests/components/energyzero/test_diagnostics.py
@@ -1,8 +1,14 @@
 """Tests for the diagnostics data provided by the EnergyZero integration."""
+from unittest.mock import MagicMock
+
 from aiohttp import ClientSession
+from energyzero import EnergyZeroNoDataError
 import pytest
 
+from homeassistant.components.homeassistant import SERVICE_UPDATE_ENTITY
+from homeassistant.const import ATTR_ENTITY_ID
 from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
 
 from tests.common import MockConfigEntry
 from tests.components.diagnostics import get_diagnostics_for_config_entry
@@ -36,3 +42,45 @@ async def test_diagnostics(
             "next_hour_price": 1.47,
         },
     }
+
+
+@pytest.mark.freeze_time("2022-12-07 15:00:00")
+async def test_diagnostics_no_gas_today(
+    hass: HomeAssistant,
+    hass_client: ClientSession,
+    mock_energyzero: MagicMock,
+    init_integration: MockConfigEntry,
+) -> None:
+    """Test diagnostics, no gas sensors available."""
+    await async_setup_component(hass, "homeassistant", {})
+    mock_energyzero.gas_prices.side_effect = EnergyZeroNoDataError
+
+    await hass.services.async_call(
+        "homeassistant",
+        SERVICE_UPDATE_ENTITY,
+        {ATTR_ENTITY_ID: ["sensor.energyzero_today_gas_current_hour_price"]},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    assert await get_diagnostics_for_config_entry(
+        hass, hass_client, init_integration
+    ) == {
+        "entry": {
+            "title": "energy",
+        },
+        "energy": {
+            "current_hour_price": 0.49,
+            "next_hour_price": 0.55,
+            "average_price": 0.37,
+            "max_price": 0.55,
+            "min_price": 0.26,
+            "highest_price_time": "2022-12-07T16:00:00+00:00",
+            "lowest_price_time": "2022-12-07T02:00:00+00:00",
+            "percentage_of_max": 89.09,
+        },
+        "gas": {
+            "current_hour_price": None,
+            "next_hour_price": None,
+        },
+    }
```